pkg/nmi/server/server.go

```diff
@@ -372,7 +384,33 @@ func (s *Server) msiHandler(w http.ResponseWriter, r *http.Request) (ns string)
 	return
 }
 
-func parseRequestHeader(r *http.Request) (podns string, podname string) {
+// Error replies to the request without the specified metadata header.
+// It does not otherwise end the request; the caller should ensure no further
+// writes are done to w.
+func metadataNotSpecifiedError(w http.ResponseWriter) {
+	metadataResp := MetadataResponse{
+		Error:            "invalid_request",
+		ErrorDescription: "Required metadata header not specified",
+	}
+	response, err := json.Marshal(metadataResp)
+	if err != nil {
+		klog.Errorf("failed to marshal metadata response, %+v", err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json; charset=utf-8")
+	w.Header().Set("Server", "Microsoft-IIS/10.0")
```

Comment:
Could we please not return IIS header, instead leave the default header returned. In the past this has helped us determine if there tokens came from IMDS or aad-pod-identity.